packages/react-components/react-virtualizer/library/src/components/Virtualizer/useVirtualizer.ts

```diff
@@ -318,25 +291,146 @@ export function useVirtualizer_unstable(props: VirtualizerProps): VirtualizerSta
+            const hOverflow = latestEntry.boundingClientRect.top - latestEntry.intersectionRect.top;
+            const hOverflowReversed = latestEntry.boundingClientRect.bottom - latestEntry.intersectionRect.bottom;
+            const wOverflow = latestEntry.boundingClientRect.left - latestEntry.intersectionRect.left;
+            const wOverflowReversed = latestEntry.boundingClientRect.right - latestEntry.intersectionRect.right;
+            const widthOverflow = reversed ? wOverflowReversed : wOverflow;
+            const heightOverflow = reversed ? hOverflowReversed : hOverflow;
+            const additionalOverflow = axis === 'vertical' ? heightOverflow : widthOverflow;
+
+            if (reversed) {
+              measurementPos += additionalOverflow;
+            } else {
+              measurementPos -= additionalOverflow;
+            }
+          } else if (latestEntry.target === beforeElementRef.current) {
+            // Get before buffers position
+            measurementPos = calculateBefore();
+            // Get exact intersection position based on overflow size (how far into window did we scroll IO?)
+            const overflowAmount =
+              axis === 'vertical' ? latestEntry.intersectionRect.height : latestEntry.intersectionRect.width;
+            // Minus from original before position
+            measurementPos -= overflowAmount;
+            // Ignore buffer size (IO offset)
+            measurementPos += bufferSize;
+
+            // Calculate how far past the window bounds we are (this will be zero if IO is within window)
+            const hOverflow = latestEntry.boundingClientRect.bottom - latestEntry.intersectionRect.bottom;
+            const hOverflowReversed = latestEntry.boundingClientRect.top - latestEntry.intersectionRect.top;
+            const wOverflow = latestEntry.boundingClientRect.right - latestEntry.intersectionRect.right;
+            const wOverflowReversed = latestEntry.boundingClientRect.left - latestEntry.intersectionRect.left;
+            const widthOverflow = reversed ? wOverflowReversed : wOverflow;
+            const heightOverflow = reversed ? hOverflowReversed : hOverflow;
+            const additionalOverflow = axis === 'vertical' ? heightOverflow : widthOverflow;
+
+            if (reversed) {
+              measurementPos += additionalOverflow;
+            } else {
+              measurementPos -= additionalOverflow;
+            }
+          }
+
+          return measurementPos;
+        };
+
+        // Get exact relative 'scrollTop' via IO values
+        const measurementPos = calculateOverBuffer();
+
+        const maxIndex = Math.max(numItems - virtualizerLength, 0);
+
+        const startIndex = getIndexFromScrollPosition(measurementPos) - bufferItems;
+
+        // Safety limits
+        const newStartIndex = Math.min(Math.max(startIndex, 0), maxIndex);
+
+        flushSync(() => {
```

Comment:
If I remember correctly, we weren't positive if we absolutely needed this? Somewhat of a nit but it would be nice if we could find out and then remove this if we don't.

Reply:
I'll be doing some more testing of heavy elements to see if this is required or not, it's possible it can help stop renders of heavy elements earlier in the render cycle once a new index is found, but will need to confirm.